Libraries/LibJS/AST.h

```diff
@@ -762,6 +762,24 @@ class ArrayExpression : public Expression {
     Vector<RefPtr<Expression>> m_elements;
 };
 
+class TemplateLiteral final : public Expression {
+public:
+    TemplateLiteral(NonnullRefPtrVector<Expression> expressions)
+        : m_expressions(expressions)
+    {
+    }
+
+    virtual Value execute(Interpreter&) const override;
+    virtual void dump(int indent) const override;
+
+    const NonnullRefPtrVector<Expression> expressions() const { return m_expressions; }
```

Comment:
Let's not return a copy of the vector. :)
```suggestion
    const NonnullRefPtrVector<Expression>& expressions() const { return m_expressions; }
```